lib/charms/ip_router_interface/v0/ip_router_interface.py

```diff
@@ -449,6 +451,46 @@ def request_network(self, networks: List[Network], custom_network_name: str = No
             str([r.name for r in ip_router_relations]),
         )
 
+    def get_routing_table(self) -> RoutingTable:
+        """Fetches combined routing tables made available by ip-router providers
+
+        Returns:
+            A list of objects of type `Network`. This list contains networks
+            from all ip-router providers that are integrated with the charm.
+        """
+        if not self.charm.unit.is_leader():
+            return
+
+        router_relations = self.model.relations.get(self.relationship_name)
+        validated_routing_table: RoutingTable = {}
+        for relation in router_relations:
+            if networks := relation.data[relation.app].get("networks"):
+                routing_table_from_databag: RoutingTable = json.loads(networks)
+                if not isinstance(json.loads(networks), dict):
+                    logger.error(
+                        "The router's routing table has been misconfigured. Can't build routing table."
+                    )
+                    return {}
+                for network_name, networks in routing_table_from_databag.items():
```

Comment:
`networks` is being redefined here, it gets confusing. Please name either here or in line 467.